make_crimes_by_street_range.R

```diff
@@ -0,0 +1,110 @@
+library(tidyverse)
+library(sf)
+library(lubridate)
+source("street_range_functions.R")
+
+# download data 
+options(timeout = 200)
+
+download.file(
+  "https://data.cincinnati-oh.gov/api/views/k59e-2pvf/rows.csv?accessType=DOWNLOAD",
+  destfile = "data/crime_incidents_rawdata.csv"
+)
+
+# read in data 
+d <- read_csv("data/crime_incidents_rawdata.csv",
+              col_types = cols(
+                DATE_REPORTED = col_datetime(format = "%m/%d/%Y %I:%M:%S %p"),
+                DATE_FROM = col_datetime(format = "%m/%d/%Y %I:%M:%S %p"),
+                DATE_TO = col_datetime(format = "%m/%d/%Y %I:%M:%S %p"),
+                CLSD = col_factor(),
+                DST = col_factor(),
+                BEAT = col_factor(),
+                LOCATION = col_factor(),
+                THEFT_CODE = col_factor(), 
+                FLOOR = col_factor(),
+                SIDE = col_factor(),
+                OPENING = col_factor(),
+                HATE_BIAS = col_factor(),
+                DAYOFWEEK = col_factor(),
+                RPT_AREA = col_factor(),
+                CPD_NEIGHBORHOOD = col_factor(),
+                WEAPONS = col_factor(),
+                DATE_OF_CLEARANCE = col_datetime(format = "%m/%d/%Y %I:%M:%S %p"),
+                VICTIM_AGE = col_factor(),
+                VICTIM_RACE = col_factor(),
+                VICTIM_ETHNICITY = col_factor(),
+                VICTIM_GENDER = col_factor(),
+                SUSPECT_AGE = col_factor(),
+                SUSPECT_RACE = col_factor(),
+                SUSPECT_ETHNICITY = col_factor(),
+                SUSPECT_GENDER = col_factor(),
+                TOTALNUMBERVICTIMS = col_integer(),
+                TOTALSUSPECTS = col_integer(),
+                UCR_GROUP = col_factor(),
+                ZIP = col_factor(),
+                COMMUNITY_COUNCIL_NEIGHBORHOOD = col_factor(),
+                SNA_NEIGHBORHOOD = col_factor()
+              ))
+
+glimpse(d)
+
+# clean up and match to coarse crime categories
+codec_category <- read_csv("crimeData_offense_codec_categories.csv") 
```

Comment:
can we change the categories to be defined in a YAML file instead of a CSV file?  It would prevent any problems with reading/writing csv files in R/excel/sharepoint.